sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/SchemaUtil.java

```diff
@@ -351,7 +351,12 @@ private BeamRowMapper(Schema schema, List<ResultSetFieldExtractor> fieldExtracto
     public Row mapRow(ResultSet rs) throws Exception {
       Row.Builder rowBuilder = Row.withSchema(schema);
       for (int i = 0; i < schema.getFieldCount(); i++) {
-        rowBuilder.addValue(fieldExtractors.get(i).extract(rs, i + 1));
+        Object value = fieldExtractors.get(i).extract(rs, i + 1);
+        if (rs.wasNull() && schema.getField(i).getType().getNullable()) {
+          rowBuilder.addValue(null);
+        } else {
+          rowBuilder.addValue(value);
```

Comment:
if the value was null and the field is not nullable, we would be setting the wrong value, right? (e.g. [for numeric columns, it would be 0](https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#getShort(int))) - should we be erroring out? perhaps we should:

```
        if (rs.wasNull()) {
          rowBuilder.addValue(null);
        } else {
          rowBuilder.addValue(value);
        }
```

what are your thoughts?

fyi @reuvenlax

Reply:
Either way seems valid to me.  The current code seems more backward-compatible, but I'm happy with either.

Reply:
Just to expand on backwards-compatibility a bit, if we added the null value to rows where the schema did not have the field as being nullable, then an exception would be thrown where it wasn't thrown before from `RowFieldMatcher.match`.  So the question is whether incorrect data is worse or newly-introduced crashes are worse.

Reply:
Incorrect data is worse. Users can fix issues with schemas if there is an error but silent data mismatch typically goes unnoticed.

Reply:
Done.